### PR TITLE
[cas] Refactor code

### PR DESCRIPTION
**Pull request by nodirg**
This change is a pure refactoring, in preparation for the commit
that adds support for Allowlist.

### CODE REVIEW

**Comment by atetubou** [comment]
Why do we need to support allowlist in addition to inputs given to upload functions?

**Comment by nodirg** [comment]
@atetubou we need to be able to upload a directory with *some*, but not all of its subpaths. Current API let's you either upload a directory entirely, or its individual pieces, but then you don't have the digest for the root.

File: go/pkg/cas/upload.go [hunk 1]
```diff
@@ -943,7 +932,7 @@ func uploadItemFromBlob(title string, blob []byte) *uploadItem {
 const pathSep = string(filepath.Separator)
 
 // joinFilePathsFast is a faster version of filepath.Join because it does not
-// call filepath.Clean.
+// call filepath.Clean. Assumes arguments are clean.
```
Comment (rubensf) [inline go/pkg/cas/upload.go:935]:
```suggestion
// call filepath.Clean. Assumes arguments are clean according to filepath.Clean specs.
```

wdyt?

**Review by rubensf — APPROVED** [review]

File: go/pkg/cas/upload.go (hunk 1 above)
Comment (nodirg) [reply go/pkg/cas/upload.go]: was incorporated.

**Pull request merged**